pkg/reconcilermanager/controllers/rootsync_controller_test.go

```diff
@@ -3141,7 +3180,9 @@ func containerResourcesMutator(overrides []v1beta1.ContainerResourcesSpec) depMu
 	return func(dep *appsv1.Deployment) {
 		for _, container := range dep.Spec.Template.Spec.Containers {
 			switch container.Name {
-			case reconcilermanager.Reconciler, reconcilermanager.GitSync, reconcilermanager.HydrationController, reconcilermanager.OciSync:
+			case reconcilermanager.Reconciler, reconcilermanager.GitSync,
```

Comment:
Having this much logic to build out test assertions (this function and the surrounding functions included) seems like a bit of a code smell, but not a big deal for this change since it's operating within the existing framework

Reply:
agreed. The test is also growing huge, and needs a refactor, but should go to a separate PR.